impeller/geometry/matrix.h

```diff
@@ -298,6 +298,13 @@ struct Matrix {
   Scalar GetMaxBasisLength() const;
 
   constexpr Scalar GetMaxBasisLengthXY() const {
+    // The full basis computation requires computing the squared scaling factor
+    // for translate/scale only matrices. This substantially limits the range of
+    // precision for small and large scales. Instead, check for the common cases
+    // and directly return the max scaling factor.
+    if (IsTranslationScaleOnly()) {
```

Comment:
Technically, we only care if [0,1] and [1,0] are 0s. The rest of the entries don't matter since we're not planning on consulting them.

Reply:
need to do this still.

Reply:
Another thing to consider, probably in a follow-on PR, there are only a couple of other uses of the non XY version of this method and they should probably all be converted to the XY version and then the non-XY method removed so it doesn't cause more problems. All of our rendering ops are 2D geometry so I don't think there is any code in the system that would want to use the regular version of this method.

Reply:
YEah that sounds like a good idea. I can take a pass after this change.

Reply:
(Done with this one though)